### DISCUSSION TITLE
Make padding and margin scales by name?

### DISCUSSION THREAD

**Discussion by pikeas** [tailwindlabs/tailwindcss#4316] [Ideas]
Tailwind does a great job at guiding users toward best practices. One are where this could be improved a bit is the spacing scale. Two ideas:

Instead of labeling the scales with numbers (p-1, p-2, etc), have you considered using names (xs, sm, md)? This would align padding and margin with the rest of TW's scales.

As it stands, this isn't possible with the number of distinct values provided. Have you considered using a restricted scale like the one in @adamwathan's own book Refactoring UI? xs through 7xl is a dozen values, which seems sufficient.

TW's docs actually give an example of replacing the defaults with a named spacing scale, which suggests that this may have been considered at some point. Nothing's coming up when searching old issues, so if this idea was previously rejected, I'd love to hear more about how that conclusion was reached.

**Comment by simonswiss** [comment]
I think that, particularly when working with Application UI type of designs, a lot of small, incremental spacing values are required. While 10-12 spacing values would work great on a specific project (a marketing website for example), we found ourselves repeatedly adding specific values in the config, project after project.

That said, nothing stops you from replacing the spacing scale with a _T-Shirt size_ scale if it helps bring consistency and remove granular decision making for the specific project you're working on.

That's the beauty of Tailwind's config file: you're encouraged to make it your own, so it can best serve your specific needs 👍
